velox/exec/tests/RowContainerTest.cpp

```diff
@@ -508,12 +555,83 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     std::vector<std::optional<T>> ascNullsFirstOrder = {
         std::nullopt, lowest, 0.0, min, max, inf, nan};
 
-    testOrderAndNullsFirstVariations<T, T>(
+    testOrderAndEqualsWithNullsFirstVariations<T>(
         type, values, ascNullsFirstOrder, [&](const auto& expectedOrder) {
           return makeNullableFlatVector<T>(expectedOrder);
         });
   }
 
+  template <typename T>
+  void populateFloatingPointTestVectors(
+      int32_t numRows,
+      std::vector<std::optional<T>>& rawData,
+      std::vector<bool>& rawExpected) {
+    TEST_FLOATING_TYPE_LIMIT_VARIABLES;
+    std::mt19937 gen(numRows);
+
+    std::vector<std::optional<T>> rawSpecialValues = {
+        std::nullopt, min, 0.0, lowest, max};
+    std::vector<bool> rawExpectedSpecialValues(rawSpecialValues.size(), false);
+    // Make sure the null value is at the beginning of the dataset.
+    // This knowledge is later used to build vectors without null values.
+    rawData.insert(
+        rawData.end(), rawSpecialValues.begin(), rawSpecialValues.end());
+    rawExpected.insert(
+        rawExpected.end(),
+        rawExpectedSpecialValues.begin(),
+        rawExpectedSpecialValues.end());
+
+    while (numRows) {
+      auto value = folly::Random::randDouble(min, max, gen);
+      // Intersperse nan values.
+      if (static_cast<int64_t>(std::fmod(value, 3.0)) == 0) {
+        rawData.push_back(nan);
+        rawExpected.push_back(true);
+      } else {
+        rawData.push_back(value);
+        rawExpected.push_back(false);
+      }
+      --numRows;
+    }
```

Comment:
Looks like this piece of code implicitly assumes that the other vector to be compared against the result of this method is an all-nan vector. But this assumption is not told in the name of this method or any comment. Could we add some comments to this method explaining what it does?